### PR TITLE
Feature/vtn 8695 source configuration

### PR DESCRIPTION
**Pull request by gitrite**
https://steel-ventures.atlassian.net/browse/VTN-8695

### CODE REVIEW

File: packages/veritone-react-common/src/components/SourceManagementForm/index.js [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+          delete contentTemplates[templateSchemaId];
+
+          return { contentTemplates };
+        });
+      }
+    } else {
+      const data = {};
+      Object.keys(templateData[templateSchemaId].definition.properties).reduce(
+        (fields, schemaDefProp) => {
+          data[schemaDefProp] =
```
Comment (charliemantle) [inline packages/veritone-react-common/src/components/SourceManagementForm/index.js]: simplify with get (see my comment in the previous PR)

File: packages/veritone-react-common/src/components/SourceManagementForm/index.js [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+  render() {
+    const { activeTab } = this.state;
+
+    return (
+      <FullScreenDialog open={this.state.openDialog}>
+        <div className={styles['sm-form-wrapper']}>
+          <ModalHeader
+            title={
+              this.state.sourceConfig.name
```
Comment (charliemantle) [inline packages/veritone-react-common/src/components/SourceManagementForm/index.js]: this.state.sourceConfig.name || 'New Source'

File: packages/veritone-react-common/src/components/SourceManagementForm/index.js [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+            )}
+            {activeTab === 1 && (
+              <ContentTemplates
+                templateData={this.props.templateData}
+                selectedTemplateSchemas={this.state.contentTemplates}
+                onListChange={this.manageTemplatesList}
+                onInputChange={this.updateTemplateDetails}
+              />
+            )}
+            <div className={styles['btn-container']}>
```
Comment (charliemantle) [inline packages/veritone-react-common/src/components/SourceManagementForm/index.js]: consistent style names

File: packages/veritone-widgets/src/widgets/SourceManagementForm/index.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import widget from '../../shared/widget';
+
+class SourceManagementFormWidget extends React.Component {
+  static propTypes = {
+    sourceTypes: arrayOf(object).isRequired,
+    templateData: objectOf(any).isRequired,
+    source: objectOf(any),
+    initialTemplates: objectOf(any),
+    onSubmit: func.isRequired,
+    onClose: func
```
Comment (charliemantle) [inline packages/veritone-widgets/src/widgets/SourceManagementForm/index.js]: This is a required prop in `SourceManagementForm` so should also be required here

File: packages/veritone-react-common/src/components/ModalHeader/index.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    icons: arrayOf(element)
+  };
+
+  static defaultProps = {
+    icons: []
+  };
+
+  render() {
+    return (
+      <div className={styles['modal-header']}>
```
[the remark this answers is not in the document]
Comment (gitrite) [reply packages/veritone-react-common/src/components/ModalHeader/index.js]: fixed

File: packages/veritone-react-common/src/components/SourceConfiguration/index.test.js [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+        id: 'audio1',
+        sourceSchema: {
+          definition: {
+            properties: {
+              url: {
+                type: 'string'
+              },
+              username: {
+                type: 'string',
+                title: 'User Name'
```
[the remark this answers is not in the document]
Comment (gitrite) [reply packages/veritone-react-common/src/components/SourceConfiguration/index.test.js]: this is for a test file though

File: packages/veritone-react-common/src/components/SourceManagementForm/index.js (hunk 2 above)
Comment (gitrite) [reply packages/veritone-react-common/src/components/SourceManagementForm/index.js]: fixed

File: packages/veritone-react-common/src/components/SourceManagementForm/index.js (hunk 3 above)
Comment (gitrite) [reply packages/veritone-react-common/src/components/SourceManagementForm/index.js]: fixed

File: packages/veritone-react-common/src/components/SourceConfiguration/index.test.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (charliemantle) [reply packages/veritone-react-common/src/components/SourceConfiguration/index.test.js]: I probably commented on the wrong file, was meant for the mock data that you're using to populate your story

File: packages/veritone-react-common/src/components/SourceManagementForm/index.js (hunk 1 above)
Comment (gitrite) [reply packages/veritone-react-common/src/components/SourceManagementForm/index.js]: fixed

File: packages/veritone-react-common/src/components/SourceManagementForm/index.js [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+        details: {
+          ...fieldValues
+        }
+      };
+    }
+
+    return this.setState(newState);
+  }
+
+  handleOnClose = () => {
```
[the remark this answers is not in the document]
Comment (gitrite) [reply packages/veritone-react-common/src/components/SourceManagementForm/index.js]: fixed

File: packages/veritone-react-common/src/components/SourceManagementForm/index.js [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+  saveConfiguration = config => {
+    return this.setState(prevState => ({
+      sourceConfig: {
+        ...prevState.sourceConfig,
+        ...config
+      }
+    }));
+  };
+
+  manageTemplatesList = (templateSchemaId, remove = false) => {
```
[the remark this answers is not in the document]
Comment (gitrite) [reply packages/veritone-react-common/src/components/SourceManagementForm/index.js]: fixed

File: packages/veritone-react-common/src/components/SourceConfiguration/index.js [hunk 9]
```diff
@@ -0,0 +1,205 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (gitrite) [reply packages/veritone-react-common/src/components/SourceConfiguration/index.js]: it does have a scrollbar. are you referring to the source management form?

[the remark this answers is not in the document]
Comment (gitrite) [reply packages/veritone-react-common/src/components/SourceConfiguration/index.js]: added scroll to source management form

File: packages/veritone-react-common/src/components/SourceConfiguration/index.test.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (gitrite) [reply packages/veritone-react-common/src/components/SourceConfiguration/index.test.js]: I think the idea behind it is to show the flexibility for the user. The title will appear just how the user specifies it

File: packages/veritone-widgets/src/widgets/SourceManagementForm/index.js (hunk 4 above)
Comment (gitrite) [reply packages/veritone-widgets/src/widgets/SourceManagementForm/index.js]: fixed shouldn't be required

**Pull request merged**